imap/jmap_mail.c

```diff
@@ -11326,9 +11326,10 @@ static void _email_create(jmap_req_t *req,
     json_t *jmailboxids = json_copy(json_object_get(jemail, "mailboxIds"));
     _append_validate_mboxids(req, jmailboxids, &parser, &have_snoozed_mboxid);
 
-    /* Validate snoozed + mailboxIds */
-    if (json_is_object(email.snoozed) && !have_snoozed_mboxid) {
+    /* Validate snoozed + mailboxIds (MUST have both or neither) */
+    if (!JNULL(email.snoozed) == !have_snoozed_mboxid) {
```

Comment:
Isn't this conditional exactly the other way round, e.g.

`email.snoozed=<some object>` and `have_snoozed_mboxid=1` results in:
`!JNULL(email.snoozed)=1` and `!have_snoozed_mboxid=0`

Reply:
> Isn't this conditional exactly the other way round, e.g.
> 
> `email.snoozed=<some object>` and `have_snoozed_mboxid=1` results in: `!JNULL(email.snoozed)=1` and `!have_snoozed_mboxid=0`

The negation is just to normalize the values to 0 or 1.  I could've used !! instead.
The logic is, if ONLY one of them is 1 then its an error.  Both 0 or both 1 is legal.

Reply:
> Isn't this conditional exactly the other way round, e.g.
> 
> `email.snoozed=<some object>` and `have_snoozed_mboxid=1` results in: `!JNULL(email.snoozed)=1` and `!have_snoozed_mboxid=0`

0 != 1 so the conditional is false and not an error.

Would be easier to understand if I change this to:
`if (!json_is_object(email.snoozed) != !have_snoozed_mboxid) {`

Reply:
Done